src/main/java/org/sagebionetworks/web/server/servlet/VersionsServlet.java

```diff
@@ -35,11 +31,14 @@ public class VersionsServlet extends HttpServlet {
       VersionsServlet.perThreadRequest.get()
     );
 
-  private final Supplier<SynapseVersionInfo> synapseVersionCache =
-    Suppliers.memoizeWithExpiration(versionSupplier(), 5, TimeUnit.MINUTES);
```

Comment:
I think this was added because John identified the web client as performing a DOS attack with version checks.  Would expiring this cache every 10 seconds be sufficient?